### PR TITLE
extend AvS model with new tags

### PR DESCRIPTION
**Pull request by i332322**
related tasks: https://github.tools.sap/kyma/backlog/issues/4251
https://github.tools.sap/kyma/backlog/issues/4372

### CODE REVIEW

File: internal/avs/external_eval_assistant.go [hunk 1]
```diff
@@ -48,8 +48,47 @@ func (eea *ExternalEvalAssistant) ProvideParentId(_ internal.ProvisioningParamet
...
+
+	Tags = append(Tags, &Tag{
+		// this returns with panic: panic: runtime error: invalid memory address or nil pointer dereference [recovered]
+		// am I referencing it wrong?
+		Content:    "string(*pp.Parameters.Provider)",
+		TagClassId: eea.avsConfig.ProviderTagClassId,
+	})
+
+	Tags = append(Tags, &Tag{
+		Content:    pp.PlatformRegion,
```
[the remark this answers is not in the document]
Comment (i332322) [reply internal/avs/external_eval_assistant.go]: ive change the order of the 2 steps within provisoningSteps. AvsInternalEval now comes after CreateRuntimeWithoutKyma step.

File: resources/kcp/charts/kyma-environment-broker/templates/secrets.yaml [hunk 2]
```diff
@@ -62,5 +60,5 @@ metadata:
 type: Opaque
 data:
   id: {{ .Values.cis.v2.id | b64enc | quote }}
-  secret: {{ .Values.cis.v2.secret | b64enc | quote }}
+  secret: {{ .Values.cis.v2secret | b64enc | quote }}
```
Comment (ebensom) [inline resources/kcp/charts/kyma-environment-broker/templates/secrets.yaml:63]:
```suggestion
  secret: {{ .Values.cis.v2.secret | b64enc | quote }}
```

File: cmd/broker/main.go [hunk 3]
```diff
@@ -745,9 +739,9 @@ func NewProvisioningProcessingQueue(ctx context.Context, provisionManager *proce
 			step:  provisioning.NewExternalEvalStep(externalEvalCreator),
 		},
 		{
-			stage:     postActionsStageName,
-			step:      provisioning.NewRuntimeTagsStep(internalEvalUpdater, provisionerClient),
-			condition: provisioning.SkipForOwnClusterPlan,
+			stage:    createRuntimeStageName,
```
Comment (ebensom) [inline cmd/broker/main.go:742]: How about to move internal evaluation step also to `postActionsStageName` ?

File: internal/avs/config.go [hunk 4]
```diff
@@ -18,16 +18,19 @@ type Config struct {
 	Disabled                                        bool   `envconfig:"default=false"`
 	InternalTesterAccessId                          int64
 	InternalTesterService                           string `envconfig:"optional"`
-	InternalTesterTags                              []*Tag `envconfig:"optional"`
 	GroupId                                         int64
 	ExternalTesterAccessId                          int64
 	ExternalTesterService                           string `envconfig:"optional"`
-	ExternalTesterTags                              []*Tag `envconfig:"optional"`
+	ExternalTesterDisabled                          bool   `envconfig:"default=false"`
```
Comment (ebensom) [inline internal/avs/config.go:24]: This is not being used, do you plan to introduce the possibility of  disabling the external avs evaluation in another PR?

File: internal/avs/external_eval_assistant.go [hunk 5]
```diff
@@ -48,8 +48,63 @@ func (eea *ExternalEvalAssistant) ProvideParentId(_ internal.ProvisioningParamet
...
+	})
+
+	Tags = append(Tags, &Tag{
+		Content:    string(operation.ProvisioningParameters.PlatformProvider),
+		TagClassId: eea.avsConfig.ProviderTagClassId,
+	})
+
+	r := ""
+	if operation.ProvisioningParameters.ErsContext.Region != nil {
+		r = *operation.ProvisioningParameters.ErsContext.Region
```
Comment (ebensom) [inline internal/avs/external_eval_assistant.go:81]: `operation.ProvisioningParameters.ErsContext.Region` doesn't hold the platform region value, you can remove it.

File: internal/avs/config.go (hunk 4 above)
Comment (i332322) [reply internal/avs/config.go:24]: the original task was not included this to be removed, only the internal/external tags and gardenerseedNameTags. if the second task (which is about to remove avs external evaulations, then in another PR i will remove this).

File: resources/kcp/charts/kyma-environment-broker/templates/deployment.yaml [hunk 6]
```diff
@@ -256,16 +256,6 @@ spec:
                 secretKeyRef:
                   key: externalTesterService
                   name: {{ .Values.avs.secretName }}
-            - name: APP_AVS_INTERNAL_TESTER_TAGS
```
[the remark this answers is not in the document]
Comment (i332322) [reply resources/kcp/charts/kyma-environment-broker/templates/deployment.yaml:259]: ack!

File: internal/avs/external_eval_assistant.go (hunk 5 above)
Comment (i332322) [reply internal/avs/external_eval_assistant.go:81]: ack!

File: cmd/broker/main.go (hunk 3 above)
Comment (i332322) [reply cmd/broker/main.go]: it is done.

**Pull request closed without merging**